### ISSUE TITLE
500 when writing to a key that exists

### ISSUE THREAD

**Issue by mitchellwrosen** [jayed/gogurl#7]
Steps to reproduce (using `httpie`):

    $ http <server> name=foo url=bar
    HTTP/1.1 200 OK
    Connection: keep-alive
    Content-Type: application/json; charset=utf-8
    Date: Wed, 14 Mar 2018 15:41:31 GMT
    Server: nginx/1.12.1
    Strict-Transport-Security: max-age=31536000; includeSubDomains
    Transfer-Encoding: chunked

    {
        "id": 23,
        "result": "success"
    }

Follwed by:

    $ http <server> name=foo url=bar
    HTTP/1.1 500 Internal Server Error
    Connection: keep-alive
    Content-Type: text/plain; charset=utf-8
    Date: Wed, 14 Mar 2018 15:43:30 GMT
    Server: nginx/1.12.1
    Strict-Transport-Security: max-age=31536000; includeSubDomains
    Transfer-Encoding: chunked

    Something went wrong

**Comment by meangrape** [comment]
This is expected, though ugly, behavior. The database returns an error when you try to create a non-unique name or url.

**Comment by mitchellwrosen** [comment]
Right. We should catch that database violation and return it as a `409 Conflict` or something, with a better error message.

**Comment by meangrape** [comment]
Working on this. I just need to figure out how to import an Exception/data constructor thingie.
